### PR TITLE
Basic search has been added to DictionaryFragment

### PR DESCRIPTION
**Pull request by sharapat**

### CODE REVIEW

File: app/src/main/java/com/shagalalab/sozlik/dictionary/DictionaryPresenter.java [hunk 1]
```diff
@@ -1,18 +1,47 @@
...
-    void searchView(String word) {
-        dictionaryView.showMessage(word);
+    void search(String word) {
+        if (Objects.equals(word, "")) {
+            dictionaryView.showError(WORD_IS_EMPTY);
+            listOfResults = new ArrayList<>();
+            dictionaryView.showResults((ArrayList<SozlikDbModel>) listOfResults);
+            return;
+        }
+        word = word.toLowerCase(Locale.ROOT);
```
[the remark this answers is not in the document]
Comment (sharapat) [reply app/src/main/java/com/shagalalab/sozlik/dictionary/DictionaryPresenter.java]: sonday dep jaz dep warning shigardig'o lint pa, findbugs ba birewi

File: app/src/main/java/com/shagalalab/sozlik/dictionary/DictionaryPresenter.java [hunk 2]
```diff
@@ -1,18 +1,47 @@
...
+        result = sozlikDao.getTranslation(word);
+        if (result != null) {
+            dictionaryView.showTranslation(result.getId());
+        } else {
+            listOfResults = sozlikDao.getSuggestions('%' + word + '%');
+            dictionaryView.showMessage(listOfResults.isEmpty() ? NOT_FOUND : FOUND);
+            dictionaryView.showResults((ArrayList<SozlikDbModel>) listOfResults);
```
[the remark this answers is not in the document]
Comment (sharapat) [reply app/src/main/java/com/shagalalab/sozlik/dictionary/DictionaryPresenter.java]: ok

File: app/src/main/java/com/shagalalab/sozlik/dictionary/DictionaryView.java [hunk 3]
```diff
@@ -1,9 +1,16 @@
...
+
+import java.util.ArrayList;
+
 /**
  * Created by QAREKEN on 3/9/2018.
  */
 
 public interface DictionaryView {
+    void showResults(ArrayList<SozlikDbModel> listOfResults);
+    void showTranslation(int translation);
```
[the remark this answers is not in the document]
Comment (sharapat) [reply app/src/main/java/com/shagalalab/sozlik/dictionary/DictionaryView.java]: he awa

File: app/src/main/java/com/shagalalab/sozlik/dictionary/DictionaryFragment.java [hunk 4]
```diff
@@ -56,8 +74,29 @@ public boolean onKey(View view, int i, KeyEvent keyEvent) {
         }
     };
 
+    @Override
+    public void showResults(ArrayList<SozlikDbModel> listOfResults) {
+        suggestionResultsAdapter = new SuggestionResultsAdapter(listOfResults, this);
+        layoutManager = new LinearLayoutManager(getActivity(), LinearLayoutManager.VERTICAL, false);
+        list.setLayoutManager(layoutManager);
+        list.setAdapter(suggestionResultsAdapter);
```
[the remark this answers is not in the document]
Comment (sharapat) [reply app/src/main/java/com/shagalalab/sozlik/dictionary/DictionaryFragment.java]: Sonda suggestionResultsAdapter'ge setListOfResults degendey metod qosiwim kerekpe?

[the remark this answers is not in the document]
Comment (sharapat) [reply app/src/main/java/com/shagalalab/sozlik/dictionary/DictionaryFragment.java]: layoutManager = new LinearLayoutManager(getActivity(), LinearLayoutManager.VERTICAL, list.setLayoutManager(layoutManager);
minani onCreate qa otkerdim

[the remark this answers is not in the document]
Comment (atabekm) [reply app/src/main/java/com/shagalalab/sozlik/dictionary/DictionaryFragment.java]: awa, `updateItems()` degen metod qossań boladı

[the remark this answers is not in the document]
Comment (sharapat) [reply app/src/main/java/com/shagalalab/sozlik/dictionary/DictionaryFragment.java]: aaa boldi tu'sinip qoydim

[the remark this answers is not in the document]
Comment (sharapat) [reply app/src/main/java/com/shagalalab/sozlik/dictionary/DictionaryFragment.java]: isledim

File: app/src/main/java/com/shagalalab/sozlik/dictionary/DictionaryFragment.java [hunk 5]
```diff
@@ -56,8 +74,29 @@ public boolean onKey(View view, int i, KeyEvent keyEvent) {
...
+
     @Override
     public void showMessage(String message) {
-        Toast.makeText(getActivity(), message, Toast.LENGTH_SHORT).show();
+        String text = "\"" + searchText.getText() + "\"" + message;
+        this.message.setText(text);
+    }
+
+    @Override
+    public void showError(String message) {
```
[the remark this answers is not in the document]
Comment (sharapat) [reply app/src/main/java/com/shagalalab/sozlik/dictionary/DictionaryFragment.java]: oshirdim

File: app/src/main/java/com/shagalalab/sozlik/dictionary/DictionaryPresenter.java [hunk 6]
```diff
@@ -1,18 +1,47 @@
...
+import java.util.Objects;
+
 /**
  * Created by QAREKEN on 3/8/2018.
  */
 
 class DictionaryPresenter {
+    private static final String FOUND = " tabılmadı, bálkim tómendegilerdi izlegen shıǵarsız.";
+    private static final String NOT_FOUND = " boyınsha nátiyjeler tabılmadı";
+    private static final String WORD_IS_EMPTY = "Hesh qanday sóz kiritpedińiz";
```
[the remark this answers is not in the document]
Comment (sharapat) [reply app/src/main/java/com/shagalalab/sozlik/dictionary/DictionaryPresenter.java]: koshirdim

File: app/src/main/java/com/shagalalab/sozlik/dictionary/DictionaryFragment.java [hunk 7]
```diff
@@ -1,31 +1,51 @@
...
-public class DictionaryFragment extends Fragment implements DictionaryView {
+public class DictionaryFragment extends Fragment implements DictionaryView, SuggestionListener {
     private DictionaryPresenter dictionaryPresenter;
     private EditText searchText;
     private Button searchButton;
+    private TextView message;
+    private RecyclerView list;
+    private SozlikDao sozlikDao;
+    private Intent intent;
+    private RecyclerView.LayoutManager layoutManager;
```
Comment (atabekm) [inline app/src/main/java/com/shagalalab/sozlik/dictionary/DictionaryFragment.java]: `sozlikDao`, `intent`, `layoutManager` úshewinde local variable'ǵa ózgertiwge boladı

File: app/src/main/java/com/shagalalab/sozlik/dictionary/SuggestionResultsAdapter.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+import java.util.ArrayList;
+
+/**
+ * Created by QAREKEN on 3/10/2018.
+ */
+
+public class SuggestionResultsAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+
+    private ArrayList<SozlikDbModel> data;
```
Comment (atabekm) [inline app/src/main/java/com/shagalalab/sozlik/dictionary/SuggestionResultsAdapter.java]: Bul List bolıwı kerek

File: app/src/main/java/com/shagalalab/sozlik/dictionary/SuggestionResultsAdapter.java [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    private SuggestionListener suggestionListener;
+
+    public SuggestionResultsAdapter(SuggestionListener suggestionListener) {
+        this.suggestionListener = suggestionListener;
+    }
+
+    void updateItems(ArrayList<SozlikDbModel> data) {
```
Comment (atabekm) [inline app/src/main/java/com/shagalalab/sozlik/dictionary/SuggestionResultsAdapter.java]: Listke ozgertiw kerek

File: app/src/main/java/com/shagalalab/sozlik/dictionary/SuggestionResultsAdapter.java [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+/**
+ * Created by QAREKEN on 3/10/2018.
+ */
+
+public class SuggestionResultsAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+
+    private ArrayList<SozlikDbModel> data;
+    private SuggestionListener suggestionListener;
+
+    public SuggestionResultsAdapter(SuggestionListener suggestionListener) {
```
Comment (atabekm) [inline app/src/main/java/com/shagalalab/sozlik/dictionary/SuggestionResultsAdapter.java]: public kerek emes

File: app/src/main/java/com/shagalalab/sozlik/dictionary/SuggestionResultsAdapter.java [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+        View view;
+        view = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_of_words_item, parent, false);
+        return new SuggestionHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(final RecyclerView.ViewHolder holder, int position) {
+        final int pos = position;
+        ((SuggestionHolder) holder).word.setText(data.get(pos).getWord());
+        ((SuggestionHolder) holder).word.setOnClickListener(new View.OnClickListener() {
```
Comment (atabekm) [inline app/src/main/java/com/shagalalab/sozlik/dictionary/SuggestionResultsAdapter.java]: `data.get(pos)` eki jerde qollanǵan, bólek variablege shıǵarǵan jaqsı

File: app/src/main/java/com/shagalalab/sozlik/dictionary/SuggestionResultsAdapter.java [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+    public int getItemCount() {
+        return data.size();
+    }
+
+    static class SuggestionHolder extends RecyclerView.ViewHolder {
+        private TextView word;
+
+        SuggestionHolder(View itemView) {
+            super(itemView);
+            word = itemView.findViewById(R.id.list_item_word);
```
Comment (atabekm) [inline app/src/main/java/com/shagalalab/sozlik/dictionary/SuggestionResultsAdapter.java]: `list_item_word` -> `item_suggestion_word`

**Pull request merged**